src/solve.h

```diff
@@ -13,6 +13,7 @@
 #define SOLVE_H
 
 #include "sat_types.h"
+#include <malloc.h>
```

Comment:
Let's move this to sat_types.h so other c files can use it. Also, what is the benefit of including just <malloc.h> rather than <stdlib.h> ?